### PR TITLE
[MP-41] Reusable Grid Items

### PR DESCRIPTION
**Pull request by thisisofficial**
### Description

Added Reusable Grid Items and it's storybook. It takes in an array of elements and shows it in a responsive grid of maximum 3 grid items per row.
image

I also needed to change some things about the previous button component ([[MP-07]](https://github.com/Touch-ofTech/touch-of-tech-storybook/pull/8)) as to make it work for my context and future ones.
It now can be customized a bit more and also the outline option changes color depending on the color of the background chosen.
image
image

### Testing Checklist

- [ ] Test cases have been added/updated
- [x] Tested locally in browser (at least one other than Chrome)
- [x] All tests pass

### What JIRA Ticket(s) does this PR address?

- [MP-41](https://cd95.atlassian.net/browse/MP-41)
- [MP-07](https://cd95.atlassian.net/browse/MP-7)

### CODE REVIEW

**Comment by thisisofficial** [comment]
[To do] There's an issue with the font in the buttons, I tried to make it work as intended but it seems it's either broken or the font isnt yet implemented on the project.
Please confirm.

File: src/components/button/Button.tsx [hunk 1]
```diff
@@ -11,6 +11,14 @@ interface ButtonProps {
    * What background color to use
    */
   backgroundColor?: string;
```
Comment (caarlosdamian) [inline src/components/button/Button.tsx:13]: add arrow.svg to icons or use one of those instead

File: src/components/button/Button.tsx [hunk 2]
```diff
@@ -58,18 +68,33 @@ export const Button = ({
   const mode = primary
     ? 'storybook-button--primary'
     : 'storybook-button--secondary';
+  if (backgroundColor) {
+    document.documentElement.style.setProperty('--custom', backgroundColor);
```
Comment (caarlosdamian) [inline src/components/button/Button.tsx:72]: dont use vanilla js here use virtual dom instead of dom directly

File: src/components/button/button.css [hunk 3]
```diff
@@ -1,47 +1,60 @@
...
+  font-weight: var(--font-weigth);
   border: 0;
   border-radius: 24px;
   cursor: pointer;
   display: inline-block;
   line-height: 1;
   min-width: 84px;
   transition: all 150ms ease-out;
+  color: white;
+  background-color: var(--primary);
```
Comment (caarlosdamian) [inline src/components/button/button.css:22]: dont need to change this css  , we have a custom button USE IT !

File: src/components/button/Button.tsx [hunk 4]
```diff
@@ -58,18 +68,33 @@ export const Button = ({
...
+  if (fontWeigth) {
+    document.documentElement.style.setProperty('--font-weigth', fontWeigth);
+  } else {
+    document.documentElement.style.setProperty('--font-weigth', '700');
+  }
 
   return (
     <button
       type="button"
       className={[
```
Comment (caarlosdamian) [inline src/components/button/Button.tsx:89]: we already have a custom button use it!

File: src/components/gridItems/GridItems.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+export const GridItems = ({ GridItemsValues }: IGridItemProps) => {
+  return (
+    <div className="grid-items-container-wrapper">
+      {GridItemsValues.map((GridItem) => (
+        <div key={GridItem.title} className="grid-item-container">
+          <div className="grid-item-title">{GridItem.title}</div>
+          <div className="grid-item-description">{GridItem.description}</div>
+          {GridItem.buttonLabel ? (
```
Comment (caarlosdamian) [inline src/components/gridItems/GridItems.tsx:21]:
```
GridItem.buttonLabel && ( <a href={GridItem.linkUrl} className="grid-item-button">
              <Button
                label={GridItem.buttonLabel}
                className="outline"
                backgroundColor="#C6754D"
                fontFamily="Montaga"
                fontWeigth="400"
                size="large"
              />{' '}
            </a> )
```

File: src/components/gridItems/GridItems.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+            <a href={GridItem.linkUrl} className="grid-item-button">
+              <Button
+                label={GridItem.buttonLabel}
+                className="outline"
+                backgroundColor="#C6754D"
+                fontFamily="Montaga"
+                fontWeigth="400"
+                size="large"
+              />{' '}
```
Comment (caarlosdamian) [inline src/components/gridItems/GridItems.tsx:30]: remove {'}

File: src/components/gridItems/GridItems.tsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+            </a>
+          ) : (
```
Comment (caarlosdamian) [inline src/components/gridItems/GridItems.tsx:32]: remove this no need to conditional render and empty space

File: src/components/gridItems/GridItems.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    title: string;
+    description: string;
+    buttonLabel?: string;
+    linkUrl?: string;
+  }[];
+}
+
+export const GridItems = ({ GridItemsValues }: IGridItemProps) => {
+  return (
+    <div className="grid-items-container-wrapper">
```
Comment (caarlosdamian) [inline src/components/gridItems/GridItems.tsx:16]: use section isntead of div 

className= 'grid-items-container'

File: src/components/gridItems/gridItems.css [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: flex-start;
+}
+
+.grid-item-title{
+    font-size: 43px;
+    height: 43px;
+    margin: 0px 0px 32px;
```
Comment (caarlosdamian) [inline src/components/gridItems/gridItems.css:35]: remove px only add 0 

`    margin: 0 0 32px;
`

File: src/components/gridItems/gridItems.css [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+.grid-item-description{
+    font-size: 18px;
+    opacity: 0.8;
+    height: 116px;
+    margin: 32px 0px;
+    text-align: left;
+}
+.grid-item-button{
+    height: 66px;
+    margin: 32px 0px 0px;
```
Comment (caarlosdamian) [inline src/components/gridItems/gridItems.css:47]: remove px only add 0

`margin:32px 0 0 `

File: src/components/gridItems/gridItems.css [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+@media screen and ( max-width: 895px){
+    .grid-items-container-wrapper{
+        grid-template-columns: 265px 265px 265px;
+    }
+}
+
+@media screen and ( max-width: 475px){
+    .grid-items-container-wrapper{
+        grid-template-columns: 300px;
+        padding: 0px 40px;
```
Comment (caarlosdamian) [inline src/components/gridItems/gridItems.css:59]: dont use px on `0 value`

File: src/components/imageCarousel/ImageSlider.stories.tsx [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+        img: 'https://s3-us-east-2.amazonaws.com/maryville/wp-content/uploads/2020/12/16162130/computer-programmer.jpg',
+      },
+      {
+        title: 'Amazing!',
+        subtitle: 'All of our savings are gone',
+        img: 'https://www.beedigital.es/wp-content/uploads/2020/09/342c4d9715440a942a78e4417fbcbad2155746109-1.jpg',
+      },
+      {
+        title: 'Erik (19)',
+        subtitle: 'Programming is fun and not stressful',
```
Comment (caarlosdamian) [inline src/components/imageCarousel/ImageSlider.stories.tsx:21]: this is not related to this ticket

File: src/components/imageCarousel/ImageSlider.tsx [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+  moveSlide,
+}: {
+  direction: 'next' | 'prev';
+  moveSlide: any;
+}) => {
+  return (
+    <button
+      type="button"
+      onClick={moveSlide}
+      className={`btn-slide ${direction}`}
```
Comment (caarlosdamian) [inline src/components/imageCarousel/ImageSlider.tsx:16]: this is not related to this ticket

File: src/components/imageCarousel/imageSlider.css [hunk 14]
```diff
@@ -0,0 +1,164 @@
+.container-slider {
+    max-width: 700px;
+    height: 500px;
+    margin: 100px auto 0;
+    position: relative;
+    overflow: hidden;
+    box-shadow: 0 10px 20px rgba(0, 0, 0, 0.2);
+  }
+  @media screen and (max-width: 700px){
```
Comment (caarlosdamian) [inline src/components/imageCarousel/imageSlider.css:9]: this is not related to this ticket

File: src/components/selectInput/selectInput.css [hunk 15]
```diff
@@ -46,6 +46,7 @@
     transition: .2s;
     transform: rotateX(0deg);
     z-index: 1;
+    filter: invert(.3) sepia(1) saturate(5) hue-rotate(175deg);
```
Comment (caarlosdamian) [inline src/components/selectInput/selectInput.css:49]: this is not related to this ticket

**Review by caarlosdamian — CHANGES_REQUESTED** [review]
check comments

File: src/components/imageCarousel/ImageSlider.stories.tsx (hunk 12 above)
Comment (thisisofficial) [reply src/components/imageCarousel/ImageSlider.stories.tsx:21]: It wasnt part of the intended pr, just a residue of the previous branch.
Let me delete this.

File: src/components/button/Button.tsx (hunk 2 above)
Comment (thisisofficial) [reply src/components/button/Button.tsx:72]: Comment dismissed: Other comments changes what this refers to.

File: src/components/button/button.css (hunk 3 above)
Comment (caarlosdamian) [reply src/components/button/button.css:20]: dont change this styles we already have custom button

File: src/components/button/button.css [hunk 16]
```diff
@@ -1,31 +1,42 @@
...
 }
 
-.storybook-button--primary {
-  color: white;
-  background-color: #1ea7fd;
+.storybook-button--secondary { 
+  border-radius: 0px;
```
Comment (caarlosdamian) [inline src/components/button/button.css:24]: again dont use 0px use 0

File: src/components/gridItems/GridItems.tsx [hunk 17]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import './gridItems.css';
```
Comment (caarlosdamian) [inline src/components/gridItems/GridItems.tsx:2]: this to line 3

File: src/components/gridItems/gridItems.css [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+    font-style: normal;
+    font-weight: 400;
+    color: #2B1105;
+}
+
+.grid-items-container{
+    width: 100%;
+    max-width: 100vw;
+    display: grid;
+    grid-template-columns: repeat(3, 30%);
```
Comment (caarlosdamian) [inline src/components/gridItems/gridItems.css:12]: use 1fr not 30% this should be responsive

File: src/components/gridItems/gridItems.css [hunk 19]
```diff
@@ -0,0 +1,68 @@
...
+    grid-template-rows: 475px;
```
Comment (caarlosdamian) [inline src/components/gridItems/gridItems.css:13]: this sould be addaptable to another items so dont use static mesuarments

**Review by caarlosdamian — CHANGES_REQUESTED** [review]
commentrs

**Review by caarlosdamian — APPROVED** [review]
